metadata_backend/helpers/validator.py

```diff
@@ -0,0 +1,85 @@
+"""Utility class for validating XML files."""
+
+import re
+import json
+from io import StringIO
+from urllib.error import URLError
+
+from aiohttp import web
+from xmlschema import XMLSchema, XMLSchemaValidationError
+from xmlschema.etree import ElementTree, ParseError
+
+from ..helpers.logger import LOG
+
+
+class XMLValidator:
+    """Validator implementation."""
+
+    def __init__(self, schema: XMLSchema, xml: str) -> None:
+        """Set Variables and initiate validation.
+
+        :param schema: Schema to be used
+        :param content: Content of XML file to be validated
+        """
+        self.schema = schema
+        self.xml_content = xml
+        self.resp_body = self.get_validation()
+        self.xmlIsValid = self.is_valid()
```

Comment:
wondering if making use of `@property` would have been useful :thinking:

Reply:
yeah I'll second this, maybe rename `get_validation() -> resp_body()` and then decorate `resp_body()` and `is_valid()` with `@property`